### PR TITLE
Alabama adjusted gross income

### LINKED ISSUE #3300: Add formula for the al_agi variable
Alabama AGI is currently not defined.

### PR DESCRIPTION
**Pull request by PavelMakarchuk**
Fixes #3300

### Summary
📄📝🧮

This pull request adds parameters and code for calculating the Alabama state adjusted gross income (AGI) for income tax purposes. It defines the income sources and adjustments that are relevant for the state tax in `income_sources.yaml` and `adjustments.yaml`, and uses them in the `al_agi` variable.

> _`IncomeTax` class_
> _adds and subtracts parameters_
> _autumn leaves fall fast_

### Walkthrough
*  Add parameters and variable for Alabama state adjusted gross income (AGI) calculation ([link](https://github.com/PolicyEngine/policyengine-us/pull/3416/files?diff=unified&w=0#diff-297f929811bc5fd7502937e9a68c63d1fa1b8e0d90622b3797fd7da99a440896R1-R20), [link](https://github.com/PolicyEngine/policyengine-us/pull/3416/files?diff=unified&w=0#diff-3b0b4a0c69cd4c557b93192d466cbbdd8cd2bf89e176f16c279c8866d9bab4afR1-R24), [link](https://github.com/PolicyEngine/policyengine-us/pull/3416/files?diff=unified&w=0#diff-8466426a1c14f7742f91c98e2122d71ec5ad64704c9b8e1b036ed1849293d208R13-R15))

### CODE REVIEW

File: policyengine_us/parameters/gov/states/al/tax/income/adjustments.yaml [hunk 1]
```diff
@@ -0,0 +1,20 @@
+description: Alabama provides these adjustments for the state adjusted gross income.
+values:
+  2021-01-01:
+  # IRA distributions prior to 1982 are not deductable
+    - roth_ira_distributions # Line 1 
+    - regular_ira_distributions # Line 1
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/al/tax/income/adjustments.yaml:6]: @MaxGhenis do we represent the Keogh RP?

source - https://www.revenue.alabama.gov/ultraviewer/viewer/basic_viewer/index.html?form=2023/01/22f40bk.pdf

image

File: policyengine_us/parameters/gov/states/al/tax/income/income_sources.yaml [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    - pension_income
+    - rental_income
+
+metadata:
+  unit: list
+  period: year
+  label: Alabama adjusted gross income income sources
+  reference:
+    - title: 2022 Alabama Department of Revenue 
+      href: https://www.revenue.alabama.gov/ultraviewer/viewer/basic_viewer/index.html?form=2023/01/22f40bk.pdf#page=8
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/al/tax/income/income_sources.yaml:20]: i dont see these income sources listed here

File: policyengine_us/parameters/gov/states/al/tax/income/income_sources.yaml [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    - title: 2021 Alabama Department of Revenue 
+      href: https://www.revenue.alabama.gov/wp-content/uploads/2022/06/21f40abk.pdf#page=7
+    - title: The Code of Alabama 1975 Section 40-18-14 
+      href: https://alisondb.legislature.state.al.us/alison/CodeOfAlabama/1975/Coatoc.htm
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/al/tax/income/income_sources.yaml:24]: this describes the income sources as comprising gross income, not AGI

File: policyengine_us/parameters/gov/states/al/tax/income/adjustments.yaml [hunk 4]
```diff
@@ -0,0 +1,20 @@
+description: Alabama provides these adjustments for the state adjusted gross income.
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/al/tax/income/adjustments.yaml:1]: This is not clear. Adjustments could be positive or negative. The law defines them as deductions.

**Review by MaxGhenis — CHANGES_REQUESTED** [review]

File: policyengine_us/parameters/gov/states/al/tax/income/income_sources.yaml (hunk 3 above)
Comment (PavelMakarchuk) [reply policyengine_us/parameters/gov/states/al/tax/income/income_sources.yaml:24]: I believe that is the income sources for the AGI are defined as gross income as described in 40-18-14.2

image

File: policyengine_us/parameters/gov/states/al/tax/income/income_sources.yaml (hunk 2 above)
Comment (PavelMakarchuk) [reply policyengine_us/parameters/gov/states/al/tax/income/income_sources.yaml:20]: Added the actual form 40

File: policyengine_us/parameters/gov/states/al/tax/income/income_sources.yaml (hunk 3 above)
Comment (PavelMakarchuk) [reply policyengine_us/parameters/gov/states/al/tax/income/income_sources.yaml]: renamed to gross_income_sources

File: policyengine_us/parameters/gov/states/al/tax/income/agi/deductions.yaml [hunk 5]
```diff
@@ -0,0 +1,20 @@
+description: Alabama provides these deductions for the state adjusted gross income.
+values:
+  2021-01-01:
+  # IRA distributions prior to 1982 are not deductable
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/al/tax/income/agi/deductions.yaml:4]:
```suggestion
  # IRA distributions prior to 1982 are not deductible.
```

**Review by MaxGhenis — APPROVED** [review]

**Pull request merged**